ui/src/app/machines/components/ActionFormWrapper/DeployForm/DeployForm.test.tsx

```diff
@@ -386,4 +382,96 @@ describe("DeployForm", () => {
     ]);
     mockUseSendAnalytics.mockRestore();
   });
+
+  it("can register a LXD VM host", () => {
+    const state = { ...initialState };
+    state.machine.selected = ["abc123"];
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <DeployForm setSelectedAction={jest.fn()} />
+        </MemoryRouter>
+      </Provider>
+    );
+    act(() =>
+      wrapper.find("Formik").props().onSubmit({
+        kernel: "",
+        oSystem: "ubuntu",
+        release: "bionic",
+        vmHostType: PodType.LXD,
+      })
+    );
+    expect(
+      store.getActions().filter((action) => action.type === "machine/deploy")
+    ).toStrictEqual([
+      {
+        type: "machine/deploy",
+        meta: {
+          model: "machine",
+          method: "action",
+        },
+        payload: {
+          params: {
+            action: NodeActions.DEPLOY,
+            extra: {
+              osystem: "ubuntu",
+              distro_series: "bionic",
+              hwe_kernel: "",
+              register_vmhost: true,
+            },
+            system_id: "abc123",
+          },
+        },
+      },
+    ]);
+  });
+
+  it("can register a libvirt KVM host", () => {
+    const state = { ...initialState };
+    state.machine.selected = ["abc123"];
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <DeployForm setSelectedAction={jest.fn()} />
+        </MemoryRouter>
+      </Provider>
+    );
+    act(() =>
+      wrapper.find("Formik").props().onSubmit({
+        kernel: "",
+        oSystem: "ubuntu",
+        release: "bionic",
+        vmHostType: PodType.VIRSH,
+      })
+    );
+    expect(
+      store.getActions().filter((action) => action.type === "machine/deploy")
+    ).toStrictEqual([
```

Comment:
Same question here about whether it should just check for the params we care about?